### ISSUE TITLE
Emacs 25.1: Symbol’s function definition is void: x-selection-owner-p

### ISSUE THREAD

**Issue by bburns** [bburns/clipmon#9]
(Transferring from other thread)

**Comment by bburns** [comment]
I set up a Xubuntu 16.04 VM and compiled Emacs 25.1 using the instructions here https://www.emacswiki.org/emacs/EmacsSnapshotAndDebian#toc3, ie

    sudo apt install autoconf automake libtool texinfo build-essential xorg-dev libgtk2.0-dev libjpeg-dev libncurses5-dev libdbus-1-dev libgif-dev libtiff-dev libm17n-dev libpng12-dev librsvg2-dev libotf-dev libxml2-dev
    cd emacs
    ./autogen.sh
    ./configure
    make bootstrap
    sudo make install

but I don't get that error - how is your system configured?

**Comment by emanresusername** [comment]
ah, ok, so i'm on a OSX and i'd been using homebrew to install/configure the cocoa app. I tried both [emacs-mac](https://github.com/railwaycat/homebrew-emacsmacport) and [emacs-plus](https://github.com/d12frosted/homebrew-emacs-plus)

when on `emacs-plus` for emacs 24.5 everything works fine, but when i switched to `emacs-mac` (emacs 25) or the latest `emacs-plus` (emacs 25). Then I get the `x-selection-owner-p` issue

**Comment by bburns** [comment]
Okay thanks - #7 was the same issue - I don't have a Mac to test with but I think I got it set up correctly - it'll be updated when Melpa next builds it.
Brian

**Comment by emanresusername** [comment]
[fixed as far as i'm concerned](http://who-is-awesome.com) #13

**Comment by bburns** [comment]
I changed the code a bit to turn off a compilation warning - it looks like this now - 

``` lisp
;; handle renamed function - x-selection-owner-p was renamed to
;; gui-selection-owner-p in emacs 25.1
(defalias 'selection-owner-p 'x-selection-owner-p)
(if (fboundp 'gui-selection-owner-p)
    (defalias 'selection-owner-p 'gui-selection-owner-p))
```

So if gui-selection-owner-p is available as in 25.1 it should use that.
